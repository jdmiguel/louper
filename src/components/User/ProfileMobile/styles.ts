import { styled } from '@mui/material/styles';

export const StyledRoot = styled('div')({
  display: 'flex',
  marginTop: 20,
  '@media (min-width: 768px)': {
    display: 'none',
  },
});

export const StyledAvatarWrapper = styled('div')({
  height: 90,
  marginRight: 20,
  position: 'relative',
  width: 90,
});
