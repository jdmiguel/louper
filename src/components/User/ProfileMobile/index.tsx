import { styled } from '@mui/material/styles';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';
import Avatar from '@mui/material/Avatar';
import { UserData } from '../../../utils/types';

const Root = styled('div')({
  display: 'flex',
  marginTop: 20,
  '@media (min-width: 768px)': {
    display: 'none',
  },
});

const AvatarWrapper = styled('div')({
  height: 90,
  marginRight: 20,
  position: 'relative',
  width: 90,
});

type Props = {
  userData: UserData;
};

const ProfileMobile = ({ userData: { avatar_url, name, login } }: Props) => (
  <Root>
    <AvatarWrapper>
      <Avatar sx={{ width: 90, height: 90, position: 'absolute' }} />
      <Avatar
        alt="user avatar"
        src={avatar_url}
        sx={{
          width: 90,
          height: 90,
          position: 'absolute',
          boxShadow: '0px 0px 14px 5px rgba(201,75,115,0.24)',
        }}
      />
    </AvatarWrapper>
    <Stack justifyContent="center">
      {name && (
        <Typography variant="h3" sx={{ marginBottom: 0.2 }}>
          {name}
        </Typography>
      )}
      <Typography variant="h4">{login}</Typography>
    </Stack>
  </Root>
);

export default ProfileMobile;
