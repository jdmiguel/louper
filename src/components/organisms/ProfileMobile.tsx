/* material-ui */
import { styled } from '@mui/material/styles';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

/* atoms */
import Avatar from '@mui/material/Avatar';

/* types */
import { UserData } from '../../utils/types';

const Root = styled('div')({
  display: 'flex',
  marginTop: 20,
  '@media (min-width: 768px)': {
    display: 'none',
  },
});

type Props = {
  userData: UserData;
};

const ProfileMobile = ({ userData: { avatar_url, name, login } }: Props) => (
  <Root>
    <Avatar
      alt="user avatar"
      src={avatar_url}
      sx={{
        width: 100,
        height: 100,
        marginRight: 2.8,
        boxShadow: '0px 0px 14px 5px rgba(201,75,115,0.24)',
      }}
    />
    <Stack justifyContent="center">
      <Typography variant="h3" sx={{ marginBottom: 0.2 }}>
        {name}
      </Typography>
      <Typography variant="h4">{login}</Typography>
    </Stack>
  </Root>
);

export default ProfileMobile;
