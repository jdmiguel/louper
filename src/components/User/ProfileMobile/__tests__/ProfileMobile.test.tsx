import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ProfileMobile from '..';

describe('<ProfileMobile />', () => {
  const props = {
    userData: {
      login: 'jdoe',
      html_url: 'https://github.com/JohnDoe',
      avatar_url: 'https://github.com/JohnDoe.jpg',
      created_at: '',
      name: 'John Doe',
      bio: '',
      email: '',
      location: '',
      blog: '',
      company: '',
      public_repos: 0,
      followers: 0,
      following: 0,
    },
  };

  it('displays the correct data', () => {
    render(<ProfileMobile {...props} />);

    const userAvatar = screen.getByAltText('user avatar');
    expect(userAvatar.getAttribute('src')).toBe('https://github.com/JohnDoe.jpg');

    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('jdoe')).toBeInTheDocument();
  });

  it('does not display the name field when no name', () => {
    render(<ProfileMobile {...props} userData={{ ...props.userData, name: null }} />);

    expect(screen.queryByText('John Doe')).not.toBeInTheDocument();
  });
});
