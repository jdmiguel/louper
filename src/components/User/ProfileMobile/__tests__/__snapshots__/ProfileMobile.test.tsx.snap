// Vitest Snapshot v1

exports[`<ProfileMobile /> > displays the content properly 1`] = `
<div>
  <div
    class="css-12acrs6"
  >
    <div
      class="css-l7hz6c"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-78bsg3-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-j6tec4-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiStack-root css-omazcq-MuiStack-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      />
    </div>
  </div>
</div>
`;
